nixos/modules/services/networking/nomad.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.nomad;
+  format = pkgs.formats.json { };
+in
+{
+  ##### interface
+  options = {
+    services.nomad = {
+      enable = mkEnableOption "Nomad, a distributed, highly available, datacenter-aware scheduler";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nomad;
+        defaultText = "pkgs.nomad";
+        description = ''
+          The package used for the Nomad agent and CLI.
+        '';
+      };
+
+      dropPrivileges = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether the nomad agent should be run as a non-root nomad user.
+        '';
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
+        description = ''
+          Configuration for Nomad. See the <link xlink:href="https://www.nomadproject.io/docs/configuration">documentation</link>
+          for supported values.
+        '';
+      };
+    };
+  };
+
+  ##### implementation
+  config = mkIf cfg.enable {
+    environment = {
+      etc."nomad.json".source = format.generate "nomad.json" cfg.settings;
+      systemPackages = [ cfg.package ];
+    };
+
+    systemd.services.nomad = {
```

Comment:
I think that we will need the ability to add packages to the `path` of the unit. Additionally there's a default set of packages that need to be there.

If the nomad service is running in client mode, then `iproute`, `iptables` and `coreutils` are necessary (and so is running with `dropPrivileges = false` though you could probably provide a list of capabilities that nomad needs if you wanted to get fancy). `coreutils` appears to already be in `PATH` when I inspect the unit.

There's also the case when a nomad client is launching tasks with the docker driver, pulling images from a private registry. Nomad supports this use case using a client configuration setting:

```
plugin "docker" {
  config {
    auth {
      # Nomad will prepend "docker-credential-" to the helper value and call
      # that script name.
      helper = "ecr-login"
    }
  }
}
```
(https://www.nomadproject.io/docs/drivers/docker#authentication)

How about a config option `extraPackages` that gets appended to the default list of `[iproute iptables]`? I could also see an argument for including `cni-plugins` by default but nomad also functions fine without those.

Reply:
> If the nomad service is running in client mode, then `iproute`, `iptables` and `coreutils` are necessary

I've added this.

> How about a config option `extraPackages`

I've added this.